File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        hold={}
        for index, num in enumerate(nums):

            find=target - num

            if find in hold and index!=hold[find]:
                if index>hold[find]:
                    return [hold[find], index]
                if index<hold[find]:
                    return [index, hold[find]]
            hold[num]=index
                
        
        return []
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {


        set<int>sorted_set(nums.begin(),nums.end());

        

        int longest =0;

        for(const auto& num : sorted_set){
            if( !sorted_set.contains(num-1)){
                int length=0;
                while(sorted_set.contains(num+length)){
                    length+=1;
                }
                if (length>longest){
                    longest=length;
                }
            }
        }

        return longest;
    }
};
